backend/src/main/java/ai/verta/modeldb/batchProcess/OwnerRoleBindingUtils.java

```diff
@@ -0,0 +1,306 @@
+    while (lowerBound < count) {
+
+      try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+        Transaction transaction = session.beginTransaction();
+
+        CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+
+        CriteriaQuery<ExperimentRunEntity> criteriaQuery =
+            criteriaBuilder.createQuery(ExperimentRunEntity.class);
+        Root<ExperimentRunEntity> root = criteriaQuery.from(ExperimentRunEntity.class);
+
+        CriteriaQuery<ExperimentRunEntity> selectQuery =
+            criteriaQuery.select(root).orderBy(criteriaBuilder.asc(root.get("id")));
+
+        TypedQuery<ExperimentRunEntity> typedQuery = session.createQuery(selectQuery);
+
+        typedQuery.setFirstResult(lowerBound);
+        typedQuery.setMaxResults(pagesize);
+        Set<String> userIds = new HashSet<>();
+        List<ExperimentRunEntity> experimentRunEntities = typedQuery.getResultList();
+        for (ExperimentRunEntity experimentRunEntity : experimentRunEntities) {
+          userIds.add(experimentRunEntity.getOwner());
+        }
+
+        LOGGER.debug("ExperimentRuns userId list : " + userIds);
+        if (userIds.size() == 0) {
+          LOGGER.warn("userIds not found for ExperimentRuns on page lower boundary {}", lowerBound);
+          lowerBound += pagesize;
+          continue;
+        }
+        Map<String, UserInfo> userInfoMap =
+            authService.getUserInfoFromAuthServer(userIds, null, null);
+        for (ExperimentRunEntity experimentRunEntity : experimentRunEntities) {
+          UserInfo userInfoValue = userInfoMap.get(experimentRunEntity.getOwner());
+          if (userInfoValue != null) {
+            try {
+              roleService.createRoleBinding(
+                  ownerRole,
+                  new CollaboratorUser(authService, userInfoValue),
+                  experimentRunEntity.getId(),
+                  ModelResourceEnum.ModelDBServiceResourceTypes.EXPERIMENT_RUN);
+            } catch (Exception e) {
+              e.printStackTrace();
+              LOGGER.error(e.getMessage());
+            }
+          } else {
+            LOGGER.error(
+                "ExperimentRun owner not found from UAC response list : ExperimentRunId - {} & userId - {}",
+                experimentRunEntity.getId(),
+                experimentRunEntity.getOwner());
+            new ModelDBException(
+                    "ExperimentRun owner not found from UAC response list : ExperimentRunId - "
+                        + experimentRunEntity.getId()
+                        + " & userId - "
+                        + experimentRunEntity.getOwner())
+                .printStackTrace();
+          }
+        }
+        LOGGER.debug("finished processing page lower boundary {}", lowerBound);
+        transaction.commit();
+        lowerBound += pagesize;
+      }
+    }
+    LOGGER.debug("ExperimentRuns migration finished");
+  }
+
+  private static void migrateDatasetVersions() {
```

Comment:
migrateExperiments, migrateExperimentRuns, migrateDatasetVersions look similar. Can they be converted to template method?

Reply:
this code will have a really short life, since its a onetime thing. I am fine if this is not templatized